router/pkg/pubsub/kafka/provider_builder.go

```diff
@@ -90,11 +106,28 @@ func buildKafkaOptions(eventSource config.KafkaEventSource) ([]kgo.Opt, error) {
 		}.AsMechanism()))
 	}
 
+	if eventSource.Authentication != nil && eventSource.Authentication.SASLSCRAM.Username != nil && eventSource.Authentication.SASLSCRAM.Password != nil && eventSource.Authentication.SASLSCRAM.Mechanism != nil {
```

Comment:
With this logic you can configure both auth methods at the same time. Is that even possible?